### ISSUE TITLE
AssertionError: Tensor with kornia.geometry.warp_perspective

### ISSUE THREAD

**Issue by mreso** [pytorch/torchdynamo#1894]
### 🐛 Describe the bug

When optimizing kornia.geometry.warp_perspective dynamo fails with "AssertError: Tensor"

Steps to reproduce:
Install dynamo and kornia:

```bash
pip install numpy --pre torch[dynamo] --force-reinstall --extra-index-url https://download.pytorch.org/whl/nightly/cu117
pip install kornia
```

Run script from minifier section

### Error logs

```
[2022-11-18 23:20:35,212] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /home/ubuntu/miniconda3/envs/td2/lib/python3.10/contextlib.py
[2022-11-18 23:20:35,212] torch._dynamo.eval_frame: [DEBUG] skipping __enter__ /home/ubuntu/miniconda3/envs/td2/lib/python3.10/contextlib.py
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [INFO] Step 1: torchdynamo start tracing func
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/torchdynamo/kornia/warp.py:13
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST (1, 2, 3, 4) []
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE UNPACK_SEQUENCE 4 [ConstantVariable(tuple)]
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST batch_size [ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST channels [ConstantVariable(int), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST height [ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST width [ConstantVariable(int)]
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/torchdynamo/kornia/warp.py:14
[2022-11-18 23:20:35,223] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL torch []
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR rand [TorchVariable(<module 'torch' from '/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/__init__.py'>)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST batch_size [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST channels [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST height [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST width [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST cuda [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST ('device',) [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(str)]
[2022-11-18 23:20:35,224] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION_KW 5 [TorchVariable(<built-in method rand of type object at 0x7ffa8423eec0>), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int), ConstantVariable(str), ConstantVariable(tuple)]
[2022-11-18 23:20:35,226] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST img_b [TensorVariable()]
[2022-11-18 23:20:35,226] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/torchdynamo/kornia/warp.py:15
[2022-11-18 23:20:35,226] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL kornia []
[2022-11-18 23:20:35,227] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR eye_like [PythonModuleVariable()]
[2022-11-18 23:20:35,227] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 3 [UserFunctionVariable()]
[2022-11-18 23:20:35,227] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST img_b [UserFunctionVariable(), ConstantVariable(int)]
[2022-11-18 23:20:35,227] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 2 [UserFunctionVariable(), ConstantVariable(int), TensorVariable()]
[2022-11-18 23:20:35,227] torch._dynamo.symbolic_convert: [DEBUG] INLINING <code object eye_like at 0x7ffa00fb87c0, file "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py", line 4>
  15           0 LOAD_FAST                0 (n)
              2 LOAD_CONST               1 (0)
              4 COMPARE_OP               1 (<=)
              6 POP_JUMP_IF_FALSE       11 (to 22)

 16           8 LOAD_GLOBAL              0 (AssertionError)
             10 LOAD_GLOBAL              1 (type)
             12 LOAD_FAST                0 (n)
             14 CALL_FUNCTION            1
             16 LOAD_FAST                0 (n)
             18 CALL_FUNCTION            2
             20 RAISE_VARARGS            1

 17     >>   22 LOAD_GLOBAL              2 (len)
             24 LOAD_FAST                1 (input)
             26 LOAD_ATTR                3 (shape)
             28 CALL_FUNCTION            1
             30 LOAD_CONST               2 (1)
             32 COMPARE_OP               0 (<)
             34 POP_JUMP_IF_FALSE       23 (to 46)

 18          36 LOAD_GLOBAL              0 (AssertionError)
             38 LOAD_FAST                1 (input)
             40 LOAD_ATTR                3 (shape)
             42 CALL_FUNCTION            1
             44 RAISE_VARARGS            1

 20     >>   46 LOAD_GLOBAL              4 (torch)
             48 LOAD_ATTR                5 (eye)
             50 LOAD_FAST                0 (n)
             52 LOAD_FAST                1 (input)
             54 LOAD_ATTR                6 (device)
             56 LOAD_FAST                1 (input)
             58 LOAD_ATTR                7 (dtype)
             60 LOAD_CONST               3 (('device', 'dtype'))
             62 CALL_FUNCTION_KW         3
             64 STORE_FAST               2 (identity)

 21          66 LOAD_FAST                2 (identity)
             68 LOAD_CONST               4 (None)
             70 BINARY_SUBSCR
             72 LOAD_METHOD              8 (repeat)
             74 LOAD_FAST                1 (input)
             76 LOAD_ATTR                3 (shape)
             78 LOAD_CONST               1 (0)
             80 BINARY_SUBSCR
             82 LOAD_CONST               2 (1)
             84 LOAD_CONST               2 (1)
             86 CALL_METHOD              3
             88 RETURN_VALUE


[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py:15
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST n []
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 0 [ConstantVariable(int)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE COMPARE_OP <= [ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE POP_JUMP_IF_FALSE 22 [ConstantVariable(bool)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py:17
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL len []
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST input [BuiltinVariable(len)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR shape [BuiltinVariable(len), TensorVariable()]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 1 [BuiltinVariable(len), ShapeVariable()]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 1 [ConstantVariable(int)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE COMPARE_OP < [ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE POP_JUMP_IF_FALSE 46 [ConstantVariable(bool)]
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py:20
[2022-11-18 23:20:35,228] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL torch []
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR eye [TorchVariable(<module 'torch' from '/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/__init__.py'>)]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST n [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>)]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST input [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int)]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR device [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int), TensorVariable()]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST input [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int), TorchVariable(cuda:0)]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR dtype [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int), TorchVariable(cuda:0), TensorVariable()]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST ('device', 'dtype') [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int), TorchVariable(cuda:0), TorchVariable(torch.float32)]
[2022-11-18 23:20:35,229] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION_KW 3 [TorchVariable(<built-in method eye of type object at 0x7ffa8423eec0>), ConstantVariable(int), TorchVariable(cuda:0), TorchVariable(torch.float32), ConstantVariable(tuple)]
[2022-11-18 23:20:35,231] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST identity [TensorVariable()]
[2022-11-18 23:20:35,231] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py:21
[2022-11-18 23:20:35,231] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST identity []
[2022-11-18 23:20:35,231] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST None [TensorVariable()]
[2022-11-18 23:20:35,231] torch._dynamo.symbolic_convert: [DEBUG] TRACE BINARY_SUBSCR None [TensorVariable(), ConstantVariable(NoneType)]
[2022-11-18 23:20:35,232] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR repeat [TensorVariable()]
[2022-11-18 23:20:35,232] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST input [GetAttrVariable(TensorVariable(), repeat)]
[2022-11-18 23:20:35,232] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR shape [GetAttrVariable(TensorVariable(), repeat), TensorVariable()]
[2022-11-18 23:20:35,233] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 0 [GetAttrVariable(TensorVariable(), repeat), ShapeVariable()]
[2022-11-18 23:20:35,233] torch._dynamo.symbolic_convert: [DEBUG] TRACE BINARY_SUBSCR None [GetAttrVariable(TensorVariable(), repeat), ShapeVariable(), ConstantVariable(int)]
[2022-11-18 23:20:35,233] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 1 [GetAttrVariable(TensorVariable(), repeat), ConstantVariable(int)]
[2022-11-18 23:20:35,233] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST 1 [GetAttrVariable(TensorVariable(), repeat), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,233] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 3 [GetAttrVariable(TensorVariable(), repeat), ConstantVariable(int), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE RETURN_VALUE None [TensorVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] DONE INLINING <code object eye_like at 0x7ffa00fb87c0, file "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/kornia/utils/misc.py", line 4>
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST H_ab [TensorVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /home/ubuntu/torchdynamo/kornia/warp.py:16
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL kornia []
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR geometry [PythonModuleVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR warp_perspective [PythonModuleVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST img_b [UserFunctionVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST H_ab [UserFunctionVariable(), TensorVariable()]
[2022-11-18 23:20:35,234] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST height [UserFunctionVariable(), TensorVariable(), TensorVariable()]
[2022-11-18 23:20:35,235] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST width [UserFunctionVariable(), TensorVariable(), TensorVariable(), ConstantVariable(int)]
[2022-11-18 23:20:35,235] torch._dynamo.symbolic_convert: [DEBUG] TRACE BUILD_TUPLE 2 [UserFunctionVariable(), TensorVariable(), TensorVariable(), ConstantVariable(int), ConstantVariable(int)]
[2022-11-18 23:20:35,235] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 3 [UserFunctionVariable(), TensorVariable(), TensorVariable(), TupleVariable()]
Traceback (most recent call last):
  File "/home/ubuntu/torchdynamo/kornia/warp.py", line 18, in <module>
    func()
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 174, in _fn
    return fn(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 286, in catch_errors
    return callback(frame, cache_size)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 476, in _convert_frame
    result = inner_convert(frame, cache_size)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 118, in _fn
    return fn(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 90, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 349, in _convert_frame_assert
    return _compile(
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 404, in _compile
    out_code = transform_code_object(code, transform)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py", line 341, in transform_code_object
    transformations(instructions, code_options)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py", line 392, in transform
    tracer.run()
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1617, in run
    super().run()
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 483, in run
    and self.step()
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 453, in step
    getattr(self, inst.opname)(inst)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 287, in wrapper
    return inner_fn(self, inst)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 910, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 395, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 196, in call_function
    return super(UserFunctionVariable, self).call_function(tx, args, kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 67, in call_function
    return tx.inline_user_function_return(
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 424, in inline_user_function_return
    result = InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1689, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py", line 1717, in inline_call_
    sub_locals, closure_cells = func.bind_args(parent, args, kwargs)
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 127, in bind_args
    tuple(map(wrap, fn.__defaults__ or [])),
  File "/home/ubuntu/miniconda3/envs/td2/lib/python3.10/site-packages/torch/_dynamo/variables/functions.py", line 34, in wrap_bound_arg
    assert isinstance(val, VariableTracker), typestr(val)
AssertionError: Tensor

from user code:
   File "/home/ubuntu/torchdynamo/kornia/warp.py", line 16, in func
    img_a = kornia.geometry.warp_perspective(img_b, H_ab, (height, width))


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Minified repro

```python
import logging
import kornia
import torch
import torch._dynamo as dynamo

torch.set_float32_matmul_precision('high')
torch._dynamo.config.verbose=True
torch._dynamo.config.log_level = logging.DEBUG

@dynamo.optimize("eager")
def func():
    batch_size, channels, height, width = 1, 2, 3, 4
    img_b = torch.rand(batch_size, channels, height, width, device='cuda')
    H_ab = kornia.eye_like(3, img_b)
    img_a = kornia.geometry.warp_perspective(img_b, H_ab, (height, width))

func()
```

**Comment by eellison** [comment]
cc @yanboliang, would you be able to take a look ?

**Comment by eellison** [comment]
@mlazos mind taking a look?

**Comment by wconstab** [comment]
This may also be fixed by [#92880](https://github.com/pytorch/pytorch/pull/92880)

Looks like @mlazos and I had a race :(
